lib/pbench/agent/tool_data_sink.py

```diff
@@ -308,111 +291,69 @@ def terminate(self):
-                    break
-                else:
-                    counter += 1
-                    if counter == 3:
-                        self.logger.error(f"{host} pmcd unreachable")
-                time.sleep(1)
-        return 1
+        pmlogger_path = find_executable("pmlogger")
+        if pmlogger_path is None:
+            pmlogger_path = self._pmlogger_path_def
+        self.pmlogger_path = pmlogger_path
+        self.USER = getpass.getuser()
 
     def launch(self):
-        global podman_path
-        pcp_remote_config = self.benchmark_run_dir / "tm" / "remote"
-        with open(pcp_remote_config, "w") as remote:
-            for host in self.host_tools_dict:
-                remote.write(
-                    f"{host} n n PCP_LOG_DIR/pmlogger/{host} -r -T24h10m -c config.{host}\n"
-                )
-
-        if not self.host_tools_dict:
+        try:
+            self.volume.mkdir()
+        except Exception as exc:
+            self.logger.error("Volume %s creation failed: '%s'", self.volume, exc)
             return 0
 
-        with open("pcp.log", "w") as pcp_logs:
-            try:
-                pcp_reg = PbenchAgentConfig(
-                    os.environ["_PBENCH_AGENT_CONFIG"]
-                ).pmlogger_reg
-            except Exception as exc:
-                self.logger.error(
-                    "Unexpected error encountered logging pbench agent configuration: '%s'",
-                    exc,
-                )
-                return 0
-
-            args = [podman_path, "pull", pcp_reg]
+        for host in self.host_tools_dict:
+            host_dir = self.volume / host
             try:
-                pcp_pull = subprocess.Popen(args, stdout=pcp_logs, stderr=pcp_logs)
-                pcp_pull.wait()
+                host_dir.mkdir()
             except Exception as exc:
-                self.logger.error("Podman pull process failed: '%s'", exc)
-                return 0
+                self.logger.error("Volume %s creation failed: '%s'", host_dir, exc)
+                continue
 
-            try:
-                os.mkdir(self.volume)
-                os.chmod(self.volume, 0o777)
-            except Exception as exc:
-                self.logger.error("Volume creation failed: '%s'", exc)
-                return 0
             args = [
-                podman_path,
-                "run",
-                "--systemd",
-                "always",
-                "-v",
-                f"{self.volume}:/var/log/pcp/pmlogger:Z",
-                "-v",
-                f"{pcp_remote_config}:/etc/pcp/pmlogger/control.d/remote:Z",
-                "--network",
-                "host",
-                "--name",
-                self.podname,
-                pcp_reg,
+                self.pmlogger_path,
+                "-r",
+                "-T",
+                "24h3s",
+                "-c",
+                "config.default",  # FIXME: per host config based on tool options
```

Comment:
This will have to be changed before remote node testing for PCP is possible. Also, is pmproxy being started anywhere yet, or is that still to be added? Note that for pmproxy, pcp-zeroconf will need to be installed in addition to base PCP in the TDS container (not sure if that is included when installing base pcp).

Reply:
> This will have to be changed before remote node testing for PCP is possible.

Not sure about that.  I believe the `config.default` will be found in the default install directory for PCP.

> Also, is pmproxy being started anywhere yet, or is that still to be added?

Still to be added.

>  Note that for pmproxy, pcp-zeroconf will need to be installed in addition to base PCP in the TDS container (not sure if that is included when installing base pcp).

We can add this later in a separate PR.

Reply:
Fixed.